yt_dlp/extractor/canalsurmas.py

```diff
@@ -0,0 +1,116 @@
+                },
+                data=json.dumps(
+                    {
+                        'username': 'canalsur_demo',
+                        'password': 'dsUBXUcI',
+                    },
+                ).encode(),
+                video_id=video_id,
+            ),
+            ('access'),
+        )
+
+        if access_token is None:
+            raise ExtractorError('Failed to retrieve a required access token.')
+
+        video_info = self._download_json(
+            url_or_request=f'https://api-rtva.interactvty.com/api/2.0/contents/content/{video_id}/',
+            video_id=video_id,
+            headers={
+                'Authorization': f'jwtok {access_token}',
+            },
+            query={
+                'optional_fields': 'description,main_category,image,duration,genre,created_at,publish_date,tags',
+            },
+        )
+
+        content_resources_linked_to_video: list | None = traverse_obj(
+            self._download_json(
+                url_or_request=f'https://api-rtva.interactvty.com/api/2.0/contents/content_resources/{video_id}/',
+                video_id=video_id,
+                headers={
+                    'Authorization': f'jwtok {access_token}',
+                },
+                query={
+                    'optional_fields': 'is_initial,media_url',
+                },
+            ),
+            ('results'),
+        )
+
+        if type(content_resources_linked_to_video) is not list:
+            raise ExtractorError('Failed to retrieve proper information about content resources linked to the specified video.')
+
+        formats = []
+
+        for content_resource_linked_to_video in content_resources_linked_to_video:
+            is_initial: bool | None = traverse_obj(
+                content_resource_linked_to_video,
+                ('is_initial'),
+            )
+
+            if is_initial:
+                media_url: str | None = traverse_obj(
+                    content_resource_linked_to_video,
+                    ('media_url'),
+                )
+
+                if type(media_url) is str:
+                    # Example URL of a M3U8 file for a specific quality (i.e. 480p): https://cdn.rtva.interactvty.com/archivo_ott/el_gran_queo/0000051496/0000051496_480p/0000051496_480p.m3u8
+                    formats = self._extract_m3u8_formats(
+                        media_url,
+                        video_id,
+                    )
+
+        return {
+            'id': video_id,
+            'title': traverse_obj(
+                video_info,
+                ('name'),
+            ),
+            'description': traverse_obj(
+                video_info,
+                ('description'),
+            ),
+            'formats': formats,
+            'thumbnail': traverse_obj(
+                video_info,
+                ('image'),
+            ),
+            'tags': traverse_obj(
+                video_info,
+                ('tags'),
+            ),
+        }
```

Comment:
The nice thing about traverse_obj is that you can combine multiple calls like so:
```suggestion
        return {
            'id': video_id,
            'formats': formats,
            'subtitles': subtitles,
            **traverse_obj(video_info, {
                'title': ('name', {str.strip}),
                'description': ('description', {str}),
                'thumbnail': ('image', {url_or_none}),
                'tags': ('tags', ..., {str}),
                'timestamp': ('created_at', {parse_iso8601}),
            }),
        }
```